alf/optimizers/optimizers.py

```diff
@@ -198,8 +199,14 @@ def step(self, closure=None):
         """
         if self._lr_scheduler is not None:
             lr = float(self._lr_scheduler())
-            for param_group in self.param_groups:
-                param_group['lr'] = lr
+            for i, (lr_scheduler, param_group) in enumerate(
+                    zip(self._lr_schedulers, self.param_groups)):
+                if lr_scheduler is not None:
+                    param_group['lr'] = lr_scheduler()
```

Comment:
If my understanding is correct, in this particular case `param_group['lr']` has already been assigned to this scheduler below at line 340 when the group is added?

Reply:
340 is only executed once when the param_group is added.
Here we need to update 'lr' according to lr_schedule for every call to step().

Reply:
Ah I see. This is calling the `scheduler` to acquire the current learning rate. Sorry about the misunderstanding.